repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -0,0 +1,360 @@
+      val whiteListURIs: Option[List[String]] =
+        for {
+          jaxbIntermediary <- Option(config.getWhiteList)
+          regexList <- Option(jaxbIntermediary.getUriRegex)
+        } yield {
+          import scala.collection.JavaConversions._
+          regexList.toList
+        }
+
+      val whiteListMatch: Boolean = whiteListURIs exists { uriList =>
+        uriList exists { pattern =>
+          logger.debug(s"checking ${request.getRequestURI} against $pattern")
+          request.getRequestURI.matches(pattern)
+        }
+      }
+
+      val authTokenValue = Option(request.getHeader(CommonHttpHeader.AUTH_TOKEN))
+
+      //Get the authenticating token!
+      val result: KeystoneV2Result = if (whiteListMatch) {
+        Pass(Map.empty[String, String])
+      } else {
+        authTokenValue map { authToken =>
+          //This block of code tries to get the token from the datastore, and provides it from real calls, if it isn't
+          val tokenValidationResult: Try[AuthResult] =
+            Option(datastore.get(s"$TOKEN_KEY_PREFIX$authToken").asInstanceOf[AuthResult]) map { validationResult =>
+              Success(validationResult)
+            } getOrElse {
+              //flatMap to unbox the Try[Try[TokenValidationResult]] so all Failure's are just packaged along
+              requestHandler.getAdminToken flatMap { adminToken =>
+                requestHandler.validateToken(adminToken, authToken) recoverWith {
+                  //Recover if the exception is an AdminTokenUnauthorizedException
+                  //This way we can specify however we want to what we want to do to retry.
+                  //Also it only retries ONCE! No loops or anything. Fails gloriously
+                  case unauth: AdminTokenUnauthorizedException =>
+                    //Clear the cache, call this method again
+                    datastore.remove(ADMIN_TOKEN_KEY)
+                    requestHandler.getAdminToken match {
+                      case Success(newAdminToken) =>
+                        requestHandler.validateToken(newAdminToken, authToken)
+                      case Failure(x) => Failure(IdentityAdminTokenException("Unable to reacquire admin token", x))
+                    }
+                }
+              }
+            }
+
+          tokenValidationResult match {
+            case Success(InvalidToken) =>
+              Reject(SC_UNAUTHORIZED)
+            case Success(validToken: ValidToken) =>
+              val uriTenantOption = requestHandler.extractTenant(request.getRequestURI)
+              val authorizedTenant = requestHandler.tenantAuthorization(uriTenantOption, validToken) match {
+                case Some(Success(tenantHeaderValues)) =>
+                  Pass(Map(OpenStackServiceHeader.TENANT_ID.toString -> tenantHeaderValues.mkString(",")))
+                case Some(Failure(e)) =>
+                  Reject(SC_UNAUTHORIZED, failure = Some(e))
+                case None =>
```

Comment:
Not sure what you're asking.

None indicates that a feature was not configured, and thus we did nothing. So if tenant authorization is not configured, we just return the previous Pass.